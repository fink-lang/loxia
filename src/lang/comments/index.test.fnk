{fink2js} = import '../../testing'
{describe, it, eq, snapshot} = import '../../testing/jest'


describe:: 'comment', fn:
  it:: 'compiles', fn:

    fink2js`
      # leading expression
      spam = ni

      ---
      Shrub ni.
      ---
      bar = fn:
        --- single line block ---
        spam

      # trailing module
    ` eq snapshot

