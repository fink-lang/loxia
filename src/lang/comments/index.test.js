import {fink2js} from '../../testing';


test('compiles assignment', ()=> {
  expect(
    fink2js(`
      # foobar spam ham
      foo = 123

      ---
      Shrub ni.
      ---
      bar = fn:
        spam
    `)
  ).toMatchSnapshot();
});
