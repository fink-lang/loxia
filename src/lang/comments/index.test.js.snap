// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles assignment 1`] = `
"/* foobar spam ham*/
const foo = 123;

/*
Shrub ni.
*/
const bar = () => spam;

Object.assign(module.exports, {
  foo,
  bar
});"
`;
